src/ophyd_async/epics/areadetector/controllers/pilatus_controller.py

```diff
@@ -45,7 +46,22 @@ async def arm(
             self._drv.num_images.set(999_999 if num == 0 else num),
             self._drv.image_mode.set(ImageMode.multiple),
         )
-        return await start_acquiring_driver_and_ensure_status(self._drv)
+
+        async def send_arm_command_and_wait_until_armed():
+            await asyncio.gather(
+                # Standard arm the detector and wait for the acquire PV to be True
+                start_acquiring_driver_and_ensure_status(self._drv),
+                # The pilatus has an additional PV that goes True when the camserver
+                # is actually ready. Should wait for that too or we risk dropping
+                # a frame
+                wait_for_value(
+                    self._drv.armed_for_triggers,
+                    True,
+                    timeout=DEFAULT_TIMEOUT,
+                ),
+            )
+
+        return AsyncStatus(send_arm_command_and_wait_until_armed())
```

Comment:
Suggest instead:
```python
            # Standard arm the detector and wait for the acquire PV to be True
            status = await start_acquiring_driver_and_ensure_status(self._drv)
            # The pilatus has an additional PV that goes True when the camserver
            # is actually ready. Should wait for that too or we risk dropping
            # a frame
            await wait_for_value(
                    self._drv.armed_for_triggers,
                    True,
                    timeout=DEFAULT_TIMEOUT,
                )
            return status
```

Reply:
This is a good suggestion. From discussion: `armed_for_triggers` should be awaited before the status is returned, because it is part of the arming process. The status returned is actually for when acquisition completes.